controllers/liveupdate_controller.go

```diff
@@ -0,0 +1,653 @@
+			sourceClusterService.Spec.Selector = selector
+			return nil
+		})
+		if err != nil {
+			r.reqLogger.Error(err, "Failed to update service")
+			return reconcile.Result{}, err
+		}
+	}
+
+	//TODO: Redirect other kind of services
+
+	// Move to next stage
+	return r.updateStage(ispnv1.LiveUpgradeRouteRedirected)
+}
+
+// syncData Copies data from the source cluster to the destination statefulSet
+func (r *LiveUpdateRequest) syncData() (ctrl.Result, error) {
+	// Obtain the podName from the new stateFul to invoke requests
+	targetStatefulSet := &appsv1.StatefulSet{}
+	err := r.Get(r.ctx, types.NamespacedName{Namespace: r.infinispan.GetNamespace(), Name: r.targetName}, targetStatefulSet)
+	if err != nil && errors.IsNotFound(err) {
+		r.reqLogger.Error(err, "Failed to find the new statefulSet")
+		return reconcile.Result{}, err
+	}
+	podList, err := r.podList(targetStatefulSet)
+
+	if err != nil {
+		r.reqLogger.Error(err, "Failed to obtain pods from the target cluster")
+		return reconcile.Result{}, err
+	}
+
+	podName := podList.Items[0].Name
+
+	names, err := r.cluster.CacheNames(podName)
+	if err != nil {
+		r.reqLogger.Error(err, "failed to get cache names from the target cluster")
+		return reconcile.Result{}, err
+	}
+
+	r.reqLogger.Info(fmt.Sprintf("Cache names in the target cluster '%s'", names))
+
+	for _, cacheName := range names {
+		count, err := r.cluster.SyncData(cacheName, podName)
+		if err != nil {
+			r.reqLogger.Error(err, fmt.Sprintf("failed to sync data for cache'%s'", cacheName))
+			return reconcile.Result{}, err
+		}
+		r.reqLogger.Info(fmt.Sprintf("Sync result %s:'", count))
+
+		err = r.cluster.DisconnectSource(cacheName, podName)
+		if err != nil {
+			r.reqLogger.Error(err, fmt.Sprintf("failed to disconnect source for cache'%s'", cacheName))
+			return reconcile.Result{}, err
+		}
+		r.reqLogger.Info("Disconnected source")
+
+	}
+	// Move to next stage
+	return r.updateStage(ispnv1.LiveUpgradeSynced)
+}
+
+// destroy Dispose the original cluster and its dependencies
+func (r *LiveUpdateRequest) destroy() (ctrl.Result, error) {
+	// Remove version label from the ping service
+	sourceClusterService := &corev1.Service{}
+	ispn := r.infinispan
+	if result, err := kube.LookupResource(ispn.Name, ispn.Namespace, sourceClusterService, r, r.log, r.eventRec, r.ctx); result != nil {
+		return *result, err
+	}
+
+	selector := sourceClusterService.Spec.Selector
+	if selector["version"] != "" {
+		_, err := controllerutil.CreateOrPatch(r.ctx, r, sourceClusterService, func() error {
+			if ispn.CreationTimestamp.IsZero() {
+				return errors.NewNotFound(schema.ParseGroupResource("infinispan.infinispan.org"), ispn.Name)
+			}
+			delete(selector, "version")
+			sourceClusterService.Spec.Selector = selector
+			return nil
+		})
+		if err != nil {
+			r.reqLogger.Error(err, "Failed to update service")
+			return reconcile.Result{}, err
+		}
+	}
+
+	// Remove version label from the pods
+	targetStatefulSet := &appsv1.StatefulSet{}
```

Comment:
Can we remove the need for this step if we update the logic so that instead of checking if `labels["version"] != ""` we instead check that if `labels["version"] == $currentVersion`.